xbmc/peripherals/bus/virtual/PeripheralBusAddon.cpp

```diff
@@ -330,14 +355,6 @@ void CPeripheralBusAddon::GetDirectory(const std::string &strPath, CFileItemList
     addon->GetDirectory(strPath, items);
 }
 
-void CPeripheralBusAddon::OnEvent(const ADDON::AddonEvent& event)
-{
-  if (typeid(event) == typeid(ADDON::AddonEvents::Enabled) ||
-      typeid(event) == typeid(ADDON::AddonEvents::Disabled) ||
-      typeid(event) == typeid(ADDON::AddonEvents::InstalledChanged))
-    UpdateAddons();
```

Comment:
This with "ADDON::AddonEvents::InstalledChanged" has not worked for updated (only install and deinstall).

A little bit of the problem of himself, but he compares installed addon id's with those already known to him.

Although the dll and the old addon are no longer present, he still thinks it would be there.

This way, the deleted dll continues nicely and the update is not used at all.